Added billing term label to purchase items.

ignaciodob opened the pull request:
As a follow-up to #51099, this adds the billing interval to "Renews on…" statuses in the purchases list.
This uses a new property on the purchase item called `billPeriodDays` to get the appropriate label.

This is an alternative approach to #51198, that had some limitations regarding Domain purchases as mentioned [here](https://github.com/Automattic/wp-calypso/pull/51198#issuecomment-1078091760)

Depends on D77601-code

Fixes: #47877

**Before** | **After**
------------ | -------------
<img width="1058" alt="Screen Shot 2021-03-18 at 2 29 22 PM" src="https://user-images.githubusercontent.com/11573483/160194094-4105408b-f027-4adb-a3e5-0c8d068704ea.png"> | <img width="1065" alt="Screen Shot 2021-03-18 at 2 26 44 PM" src="https://user-images.githubusercontent.com/11573483/160194075-5f141efd-24e2-4308-a7b4-d8ce52497eaf.png">



**To test:**
- On an account with purchases (with payment methods assigned and auto-renew on)
- Visit Me > Purchases
- Verify that the purchase status includes a billing interval
- Visit My Site > Upgrades > Purchases
- Verify that the purchase status includes a billing interval

Commit:
Added billing term label to purchase items.

Based on new purchase property called bill_period_days

Commit:
Fixed getByPurchaseId test

Added new billPeriodDays purchase property

sirbrillig reviewed (approved):
Works as described! I tested in other languages and it also looks good.
<img width="359" alt="Screen Shot 2022-03-28 at 11 54 09 AM" src="https://user-images.githubusercontent.com/2036909/160438420-8a2f6969-bb73-459b-9a2b-d7b6540cdf0a.png">

danielmx-dev reviewed (changes requested):
I tested this using the ES locale and found something odd. 

This is how the purchases list is displayed in calypso.live:

![image](https://user-images.githubusercontent.com/15204776/160459321-268c5652-a728-46b2-9d1e-39446c738230.png)


If you notice, the translations being used for monthly and yearly are "mensual" and "anual", which are technically correct but are not appropriate for the context (the appropriate translations would be "mensualmente" and "anualmente"). To solve this, I suggest creating separate strings for each term so they can be translated properly for each language, instead of interpolating the interval in the existing string.

@sirbrillig Do you also agree with this?

sirbrillig commented:
That sounds fine to me. I did worry about that a little bit. Thank you for checking in more languages than I did.

Commit:
Split purchase billing term strings for better translations

Previously with the interpolation, we got wrong translations in specific locales like ES. Splitting the strings into individual ones should give us a correct translation in other locales.

danielmx-dev commented on client/me/purchases/purchase-item/index.jsx:
```diff
@@ -196,6 +202,52 @@ class PurchaseItem extends Component {
 			}
 
+			if ( purchase.billPeriodDays ) {
+				const translateOptions = {
+					args: {
+						amount: purchase.priceText,
+						date: renewDate.format( 'LL' ),
+					},
+					components: {
+						span: <span className="purchase-item__date" />,
+					},
+				};
```
I wasn't sure about this pattern of reusing the translate options object but it seems it has been already used in other places ( eg. https://github.com/Automattic/wp-calypso/blob/5152f5f056421d148081a8de3514edd412d833a1/packages/wpcom-checkout/src/checkout-line-items.tsx#L613-L623 ), so it should be fine to also use it here.

danielmx-dev reviewed (approved):
The code looks good and works as expected! 💯

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7209483

Thank you @ignaciodob for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7209483).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7209483).
